chips/litex/src/uart.rs

```diff
@@ -344,12 +344,12 @@ impl<'a, R: LiteXSoCRegisterConfiguration> uart::Transmit<'a> for LiteXUart<'a,
         //
         // If it does not fill the FIFO, an
         // interrupt will _not_ be generated and hence we have to
-        // perform the upcall using a deferred call.
+        // perform the callback using a deferred call.
         //
         // If we fill up the FIFO, an interrupt _will_ be
         // generated. We can transmit the rest using `resume_tx` and
         // directly call the callback there, as we are guaranteed to
-        // be in an upcall.
+        // be in a callback.
```

Comment:
Shouldn't this stay as upcall? for tock's rules about scheduling callbacks, I think what matters is that the callback is only called from within an upcall

Reply:
I think callback is better here, given that this references an "upcall" from the scheduler, right? That shouldn't be confused with an upcall to userspace. The important thing here is that we're in a different stack frame to be able to safely call `resume_tx`, such that it can enter it's resources again. This is actually the exact issue that confused me in the initial discussion for this in the first place. I suppose saying "upcall from the scheduler" could be a valid option? For reference, this is @alevy's comment which made me call this an "upcall": https://github.com/tock/tock/issues/1736#issuecomment-610015515 :smile: